tests/js/spec/components/bulkNotice.spec.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+
+import {shallow} from 'sentry-test/enzyme';
+
+import BulkNotice, {
+  getEverythingSelectedText,
+  getSelectEverythingText,
+} from 'app/components/bulkNotice';
+
+const props = {
+  allRowsCount: 64,
+  selectedRowsCount: 10,
+  onCancelAllRows: () => {},
+  onSelectAllRows: () => {},
+  columnsCount: 4,
+  isPageSelected: false,
+  isEverythingSelected: false,
+};
+
+describe('BulkNotice', function() {
+  it('does not render if the whole page is not selected', function() {
+    const wrapper = shallow(<BulkNotice {...props} />);
+    expect(wrapper.isEmptyRender()).toBe(true);
+  });
+
+  it('shows the right page summary', function() {
+    const wrapper = shallow(<BulkNotice {...props} isPageSelected />);
+    expect(wrapper.text()).toBe(
+      `${props.selectedRowsCount} items on this page selected. Select all ${props.allRowsCount} items.`
+    );
+  });
+
+  it('can select all rows across all pages', function() {
+    const onSelectAllRows = jest.fn();
+    const wrapper = shallow(
+      <BulkNotice {...props} isPageSelected onSelectAllRows={onSelectAllRows} />
+    );
+    wrapper.find('a').simulate('click');
+    expect(onSelectAllRows).toHaveBeenCalled();
+  });
+
+  it('can deselect all once everything is selected', function() {
+    const onCancelAllRows = jest.fn();
+    const wrapper = shallow(
+      <BulkNotice
+        {...props}
+        isPageSelected
+        isEverythingSelected
+        onCancelAllRows={onCancelAllRows}
```

Comment:
onCancel is fine, but I think it would make more sense if it was:

```suggestion
        onDeselectAllRows={onDeselectAllRows}
```